sources/editor/Stride.Assets.Presentation/CurveEditor/ViewModels/RotationCurveViewModel.cs

```diff
@@ -204,26 +204,26 @@ protected internal override void UpdateMaxMin()
                 MaxY = maxy;
         }
 
-        /// <summary>
-        /// This method should have better precision than <see cref="Matrix.DecomposeXYZ(out Vector3)"/>, especially on the y-axis.
-        /// </summary>
-        /// <param name="matrix"></param>
-        /// <param name="rotation"></param>
-        private static void DecomposeXYZ(Matrix matrix, out Vector3 rotation)
-        {
-            var rotY = Math.Abs(matrix.M13) - 1 < 1e-6f ? Math.Acos(Math.Max(Math.Abs(matrix.M11), Math.Abs(matrix.M33))) : Math.Asin(-matrix.M13);
-            rotation.Y = (float)(MathF.Sign(-matrix.M13)*rotY);
-            var test = Math.Cos(rotY);
-            if (test > 1e-6f)
-            {
-                rotation.Z = MathF.Atan2(matrix.M12, matrix.M11);
-                rotation.X = MathF.Atan2(matrix.M23, matrix.M33);
-            }
-            else
-            {
-                rotation.Z = MathF.Atan2(-matrix.M21, matrix.M31);
-                rotation.X = 0.0f;
-            }
-        }
+        ///// <summary>
+        ///// This method should have better precision than <see cref="Matrix.DecomposeXYZ(out Vector3)"/>, especially on the y-axis.
+        ///// </summary>
+        ///// <param name="matrix"></param>
+        ///// <param name="rotation"></param>
+        //private static void DecomposeXYZ(Matrix matrix, out Vector3 rotation)
```

Comment:
Can you test if this code provides any significant difference in output values from the new one? If not, it should be removed instead of leaving it commented. If yes, we should try to understand what is the difference here.

Reply:
Reinstated but fixed the edge case values. Thanks for making me double check.

So the `M13` test in `RotationCurveViewModel` was already testing the +1 case properly (`Math.Abs(matrix.M13) - 1 < 1e-6f` is exactly the same as doing `MathUtil.IsOne(Math.Abs(M13)`).

So doc was right in that the yaw value can be more precise. This is because it does `Math.Acos(Math.Max(Math.Abs(matrix.M11), Math.Abs(matrix.M33)))` to get the yaw value as it approaches the edge case (eg. when the yaw is close to Pi/2) whereas the Matrix's code just sets it straight to `+/-PiOverTwo` (this is both the old version, and in Eberly's solution). I tested angles approaching Pi/2 (and a bit past it), and it does also only hit the edge case exactly on Pi/2 whereas the Matrix's code would treat the angle as an edge case as it got closer to Pi/2.
For the sake of consistency with the old code (and performance), I think the Matrix version can still just set it to PiOverTwo instead of the additional Acos/Max/Abs calls.